lib/shopify_api/rest/draft_order.ex

```diff
@@ -0,0 +1,139 @@
+      %{
+        draft_order: %{
+          id: 994118539,
+          note: "Customer contacted us about a custom engraving on the item"
+        }
+      }
+  """
+  def update(%AuthToken{} = auth, %{draft_order: %{id: draft_order_id}} = draft_order),
+    do: Request.put(auth, "draft_orders/#{draft_order_id}.json", draft_order)
+
+  @doc """
+  Retrieve a list of all draft orders
+
+  ## Example
+
+      iex> ShopifyAPI.REST.DraftOrder.all(auth)
+      {:ok, %{"draft_orders" => []}}
+  """
+  def all(%AuthToken{} = auth), do: Request.get(auth, "draft_orders.json")
+
+  @doc """
+  Retrieve a specific draft order
+
+  ## Example
+
+      iex> ShopifyAPI.REST.DraftOrder.get(auth, integer)
+      {:ok, %{"draft_order" => %{...}}}
+  """
+  def get(%AuthToken{} = auth, draft_order_id),
+    do: Request.get(auth, "draft_orders/#{draft_order_id}.json")
+
+  @doc """
+  Retrieve a count of all draft orders
+
+  ## Example
+
+      iex> ShopifyAPI.REST.DraftOrder.count(auth)
+      {:ok, %{count: integer}}
+  """
+  def count(%AuthToken{} = auth), do: Request.get(auth, "draft_orders/count.json")
+
+  @doc """
+  Send an invoice for a draft order
+
+  ## Example
+
+      iex> ShopifyAPI.REST.DraftOrder.send_invoice(auth, %{draft_order_invoice: %{}})
+      {:ok, %{"draft_order_invoice" => %{...}}}
+
+      To send the default invoice:
+      %{
+        draft_order_invoice: %{}
+      }
+
+      To send a customized invoice:
+      %{
+        draft_order_invoice: %{
+          to: "first@example.com",
+          from: "steve@apple.com",
+          bcc: [
+            "steve@apple.com"
+          ],
+          subject: "Apple Computer Invoice",
+          custom_message: "Thank you for ordering!"
+        }
+      }
+  """
+  def send_invoice(
+        %AuthToken{} = auth,
+        draft_order_id,
+        %{draft_order_invoice: %{}} = draft_order_invoice
+      ),
+      do:
+        Request.post(
+          auth,
+          "draft_orders/#{draft_order_id}/send_invoice.json",
+          draft_order_invoice
+        )
+
+  @doc """
+  Delete a draft order
+
+  ## Example
+
+      iex> ShopifyAPI.REST.DraftOrder.delete(auth, integer)
+      {:ok, %{}}
+  """
+  def delete(%AuthToken{} = auth, draft_order_id),
+    do: Request.delete(auth, "draft_orders/#{draft_order_id}.json")
+
+  @doc """
+  Complete a draft order
+
+  ## Example
+
+      To complete a draft order, marking it as paid:
+      iex> ShopifyAPI.REST.DraftOrder.complete(auth, integer)
+
+      To complete a draft order, marking it as pending:
+      iex> ShopifyAPI.REST.DraftOrder.complete(auth, integer, true)
+
+      {:ok, %{"draft_order", %{...}}}
+  """
+  def complete(%AuthToken{} = auth, draft_order_id),
+    do: Request.put(auth, "draft_orders/#{draft_order_id}/complete.json", %{})
+
+  def complete(%AuthToken{} = auth, draft_order_id, true),
+    do:
+      Request.put(auth, "draft_orders/#{draft_order_id}/complete.json?payment_pending=true", %{})
```

Comment:
This should probably just be merged with the one above and allow for filter params to be passed in.